compiler/rustc_attr/src/builtin.rs

```diff
@@ -856,177 +856,180 @@ impl IntType {
 /// structure layout, `packed` to remove padding, and `transparent` to delegate representation
 /// concerns to the only non-ZST field.
 pub fn find_repr_attrs(sess: &Session, attr: &Attribute) -> Vec<ReprAttr> {
-    use ReprAttr::*;
+    if attr.has_name(sym::repr) { parse_repr_attr(sess, attr) } else { Vec::new() }
+}
 
+pub fn parse_repr_attr(sess: &Session, attr: &Attribute) -> Vec<ReprAttr> {
+    assert!(attr.has_name(sym::repr), "expected a a `#[repr(..)]`, found: {:?}", attr);
```

Comment:
```suggestion
    assert!(attr.has_name(sym::repr), "expected `#[repr(..)]`, found: {:?}", attr);
```
before merging. Don't want to do it during the perf run.